src/components/date/date.pw.tsx

```diff
@@ -0,0 +1,1349 @@
+    );
+    await expect(dayPickerNavButton2).toHaveCSS(
+      "outline",
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+  });
+});
+
+test.describe("Events tests", () => {
+  test(`should call onChange callback when a clear event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DateInputCustom
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const input = getDataElementByValue(page, "input");
+    await input.clear();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onChange callback when a type event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <DateInputCustom
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const input = getDataElementByValue(page, "input");
+    await input.fill("1");
+    expect(callbackCount).toBe(1);
+  });
+
+  ([
+    ["en-US", "05/12/2022", "enUS"],
+    ["en-CA", "05/12/2022", "enCA"],
+    ["en-ZA", "12/05/2022", "enZA"],
+    ["de", "12.05.2022", "de"],
+    ["es-ES", "12/05/2022", "es"],
+    ["fr-FR", "12/05/2022", "fr"],
+    ["fr-CA", "12/05/2022", "frCA"],
+    ["zh-CN", "2022/05/12", "zhCN"],
+    ["pl-PL", "12.05.2022", "pl"],
+    ["bg-BG", "12.05.2022", "bg"],
+    ["zh-HK", "12/05/2022", "zhHK"],
+    ["hu-HU", "2022. 05. 12.", "hu"],
+    ["fi-FI", "12.05.2022", "fi"],
+    ["de-AT", "12.05.2022", "deAT"],
+    ["ko-KR", "2022. 05. 12.", "ko"],
+    ["ar-EG", "12/05/2022", "arEG"],
+    ["hi-HI", "12/05/2022", "hi"],
+    ["sl-SI", "12. 05. 2022", "sl"],
+    ["lv", "12.05.2022.", "lv"],
+  ] as [string, string, string][]).forEach(
+    ([localeValue, formattedValueParam, dateFnsLocaleKey]) => {
+      test(`should use ${localeValue} locale and change the formattedValue to ${formattedValueParam} after selecting date`, async ({
+        mount,
+        page,
+      }) => {
+        let callbackCount = 0;
+        await mount<HooksConfig>(
+          <DateWithLocales
+            onChange={() => {
+              callbackCount += 1;
+            }}
+          />,
+          {
+            hooksConfig: { localeName: dateFnsLocaleKey },
+          }
+        );
+
+        const getDatetoEnter = () => {
+          if (["en-US", "en-CA"].includes(localeValue))
+            return MMDDYYYY_DATE_TO_ENTER;
+          if (["zh-CN", "hu-HU", "ko-KR"].includes(localeValue))
+            return YYYYMMDD_DATE_TO_ENTER;
+          return DDMMYYY_DATE_TO_ENTER;
+        };
+
+        const input = getDataElementByValue(page, "input");
+        await input.clear();
+        await input.fill(getDatetoEnter());
+        await input.click();
+
+        const pickerByText = page
+          .locator(".DayPicker-Day")
+          .filter({ hasText: "12" });
+        await pickerByText.click();
+        await expect(input).toHaveValue(formattedValueParam);
+
+        expect(callbackCount).toBe(4);
```

Comment:
**question** (non-blocking): obviously the test passes but I'm not sure why the callback count is exactly 4 - my understanding is that it gets called on each keypress, and since the "date to enter" has many digits, I'd expect the call count to be a lot bigger than 4.

Regardless of whether this is correct or not, I'm not sure the precise number of calls matters here in a complex scenario like this. I'd rather keep this test just for the formatted value, and for checking `onChange` is called do a test with a single call for each scenario we want it called for.

I'd make similar comments on the next 2 groups of tests as well.